julearn/pipeline/pipeline_creator.py

```diff
@@ -212,6 +214,17 @@ def add(
         # Validate the step
         self._validate_step(step, apply_to)
 
+        # Check that if the name is already in the steps, we are adding them
+        # at the same place.
+        if name is not None and name in [x.name for x in self._steps]:
```

Comment:
[Consideration, I will accept the PR even if decided against this]
Again maybe something else than x ? e.g. i ? 
I also see that at other points. I will stop mentioning it as either you agree and adjust everywhere or these comments just become obnoxious

Reply:
I'm changing them. This is about maintainability/readability. So I agree with you.